UpstreamRepoPath needs a valid Go module path but does not make this clear

### What happened?

While attempting to override the upstream docs location for a provider, I tried pointing the `UpstreamRepoPath` field to a valid github repository address:
```
UpstreamRepoPath: "github.com/F5Networks/terraform-provider-bigip",
```
And was told by the bridge that it could not find any docs for any resource.
The inline documentation for this field reads:
```
// An optional path that overrides upstream location during docs lookup
```
which lends itself to the assumption to simply use a repo path on github.

When getting some debug prints from `pkg/tfgen/source.go` it turns out that `UpstreamRepoPath` is effectively assumed to be set to either `./upstream` (in the case of a submodule in the provider) or `""`(unset). In the latter case, a valid go mod path is generated in the bridge from the upstream repo name. It looked as follows for me:
`/Users/guin/go/pkg/mod/github.com/!f5!networks/terraform-provider-bigip@v1.22.5`.

The bridge should either:
- be able to generate a valid value for this field from a valid upstream repo name
- document this field to make it clear that this is a (local?) go mod path, not a github path.


### Example

Set any provider's `UpstreamRepoPath` to its upstream terraform provider's github repo, and watch `tfgen` fail.

```
...
* bigip_waf_policy: could not find docs for resource 'bigip_waf_policy'. Override the Docs property in the resource mapping. See type tfbridge.DocInfo for details.

make: *** [tfgen_no_deps] Error 255
```

### Output of `pulumi about`

n/a

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/pkg/tfbridge/info/info.go
+++ b/pkg/tfbridge/info/info.go
@@ -119,7 +119,18 @@ type Provider struct {
 	// Rules that control file discovery and edits for any subset of docs in a provider.
 	DocRules *DocRule
 
-	UpstreamRepoPath string // An optional path that overrides upstream location during docs lookup
+	// An optional file path to the root of the upstream provider's git repo, for use in docs generation.
+	//
+	// It UpstreamRepoPath is left blank, it is inferred to the location where Go downloaded the build
+	// dependency of the provider. The following fields influence the inference decision:
+	//
+	// - GitHubHost
+	// - GitHubOrg
+	// - Name
+	// - TFProviderModuleVersion
+	//
+	// This list may change over time.
+	UpstreamRepoPath string
 
 	// EXPERIMENTAL: the signature may change in minor releases.
 	//

--- a/pkg/tfbridge/info/info.go
+++ b/pkg/tfbridge/info/info.go
@@ -121,7 +121,7 @@ type Provider struct {
 
 	// An optional file path to the root of the upstream provider's git repo, for use in docs generation.
 	//
-	// It UpstreamRepoPath is left blank, it is inferred to the location where Go downloaded the build
+	// If UpstreamRepoPath is left blank, it is inferred to the location where Go downloaded the build
 	// dependency of the provider. The following fields influence the inference decision:
 	//
 	// - GitHubHost

